lib/assignDisabledRanges.mjs

```diff
@@ -268,34 +345,58 @@ export default function assignDisabledRanges(root, result) {
 				continue;
 			}
 
-			throw comment.error(`"${ruleToEnable}" has not been disabled`, {
+			throw node.error(`"${ruleToEnable}" has not been disabled`, {
 				plugin: 'stylelint',
 			});
 		}
 	}
 
 	/**
-	 * @param {PostcssComment} comment
+	 * @param {PostcssNode} node
+	 * @param {string} part
+	 * @param {number} offset
+	 */
+	function checkCommentContainingNode(node, part, offset) {
+		tokenize({ css: part })
+			.filter((token) => {
+				return isTokenComment(token);
+			})
+			.forEach((token) => {
+				const [, text, start, end] = token;
```

Comment:
[suggest] Minor. For the performance, reducing the loop count may be a bit more effective:

```suggestion
		tokenize({ css: part })
			.forEach((token) => {
				if (!isTokenComment(token)) return;

				const [, text, start, end] = token;
```